### PR TITLE
Recast column types.

### PR DESCRIPTION
**Pull request by mialy-defelice**
When updating tables on synapse the function `update_df` is called in `df_utils.py`. This function invokes a native pandas function called `df.update`. There is a known issue with pandas where sometimes upon calling `df.update` the column types can sometimes be changed, for example, np.int64 to np.float64. This can cause real issues when uploading to Synapse when the column type is expecting a certain type.

This PR addresses this issue by recasting the column type based on the input table.

### CODE REVIEW

File: schematic/utils/df_utils.py [hunk 1]
```diff
@@ -77,6 +77,10 @@ def update_df(
     input_df_idx.reset_index(inplace=True)
     input_df_idx = input_df_idx[input_df.columns]
 
+     # Sometimes pandas update can change the column datatype, recast
+    for col in input_df_idx.columns:
+        input_df_idx[col] = input_df_idx[col].astype(input_df.dtypes[col])
```
Comment (milen-sage) [inline schematic/utils/df_utils.py:82]: @mialy-defelice just confirming I understand correctly, the dtypes gets the numpy type of data in the given column and astype ensures pandas sets up the type accordingly in the dataframe?

Comment (mialy-defelice) [reply schematic/utils/df_utils.py:82]: @milen-sage It was converting int to float. So yes at the end of the transformations, I just use as type to reset the type to the column types in the input table.

Comment (mialy-defelice) [reply schematic/utils/df_utils.py:82]: I'll go through later and see why all those tests are failing.

Comment (milen-sage) [reply schematic/utils/df_utils.py:82]: got it

**Review by milen-sage — APPROVED** [review]

**Review by milen-sage — CHANGES_REQUESTED** [review]
Just a quick change - could you take a look at the [test](https://github.com/Sage-Bionetworks/schematic/runs/4859491698?check_suite_focus=true#step:8:94) for this function it looks like it might be straightforward to update it to accommodate the type cast?

**Comment by mialy-defelice** [comment]
Hey @milen-sage ,
Just to make sure I'm updating the test correct, you are fine with me updating the test to match what we would now expect? It seems like the test is expecting a type change.

**Comment by milen-sage** [comment]
@mialy-defelice yes let's try making the test expect what the updated code outputs.

**Comment by mialy-defelice** [comment]
@milen-sage  I updated the test so now its passing all its checks.

**Review by milen-sage — APPROVED** [review]

**Comment by milen-sage** [comment]
@mialy-defelice feel free to merge!

**Pull request merged**